### ISSUE TITLE
Keep a CHANGELOG

### ISSUE THREAD

**Issue by G-Rath** [shellscape/webpack-plugin-serve#43]
Documentation Is:

- [x] Missing
- [x] Needed
- [ ] Confusing
- [ ] Not Sure?

### Please Explain in Detail...
Changelogs are useful for knowing what changed in a... log?

But seriously: It's be awesome if a changelog was used to track changes between releases, as it makes things easier for developers to find out what to expect when comparing versions, without having to read through all the commit logs.

### Your Proposal for Changes

[Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

**Comment by shellscape** [comment]
We'll be updating the [releases](https://github.com/shellscape/webpack-plugin-serve/releases) page for the next version. 🍺

**Comment by G-Rath** [comment]
@shellscape while that's a nice start, personally I prefer having a `CHANGELOG.md`, as releases is a `github` feature; meanwhile, a `CHANGELOG` file will follow the code around, until it's explicitly left behind.

In particular, this means `CHANGELOG` usually ends up being published in the npm package, and so is available with each version of the package (as well as being available offline).

**Comment by shellscape** [comment]
Thanks for explaining. My personal preference is different. While we might be able to automate a changelog from the releases, I can pretty much guarantee it won't ship with the NPM package :smile: I like to keep packages as lean as humanly possible.

**Comment by shellscape** [comment]
Have given this a great deal of thought (as all suggestions for change in direction should!) and I still prefer the releases tab. We use https://www.conventionalcommits.org/en/v1.0.0-beta.2/ style commit messages (which is also what Angular uses) along with tagged npm versions and squashed commits - so either using `git log` or Github's commit history would be just as effective as a Changelog in that way. It took me a while to find, but [this comment](https://github.com/sindresorhus/module-requests/issues/106#issuecomment-342012360) perfectly summarizes why I prefer the Releases tab over a CHANGELOG artifact.

**Comment by G-Rath** [comment]
@shellscape I just remembered this, which you might find interesting:

> I can pretty much guarantee it won't ship with the NPM package

I can guarantee that it would, since `npm` [*always* includes `CHANGELOG`](https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package) if it exists, regardless of your settings:

> The following paths and files are never ignored, so adding them to .npmignore is pointless:
> * package.json
> * README (and its variants)
> * CHANGELOG (and its variants)
> * LICENSE / LICENCE

To me, this indicates that `npm` feels `CHANGELOG` is something that should be distributed with the rest of the package, which is something I agree with.

The releases tab is a github-only feature, and not accessible offline - say for example you removed your github repo (maybe a left-pad kind of situation, who knows, but it could technically happen). If you have an application that's dependent on an old version of that package, and you decide to update that dependency, by having the CHANGELOG bundled with every version of the package you'd at least have a chance of figuring out everything you've got to do for that update, even tho the original repo is gone.

It's a somewhat extreme, but still relevant (of which left-pad is a great example) argument.

Git is great, when its there - but once a package is bundled & on npm it's out of the git infrastructure. It's also great to keep packages lean, but CHANGELOGs are typically really small compared to the rest of the package.

---- 
I understand both sides of the fence, and so I'm happy so long as changes are recorded *somewhere* sane, but I stumbled across the npm link while setting up a new package of my own for publishing, and was like "omg why didn't I remember this half a year ago when talking with Andrew about changelogs!?"
